pkg/providers/github.go

```diff
@@ -69,14 +69,35 @@ func (g *gitHub) Fetch() (*File, error) {
 // returns the corresponding name and url to fetch the version
 func (g *gitHub) GetLatestVersion() (string, string, error) {
 	log.Debugf("Getting latest release for %s/%s", g.owner, g.repo)
-	release, _, err := g.client.Repositories.GetLatestRelease(context.TODO(), g.owner, g.repo)
+	release, err := g.getAnyLatestRelease(context.TODO())
 	if err != nil {
 		return "", "", err
 	}
 
 	return release.GetTagName(), release.GetHTMLURL(), nil
 }
 
+func (g *gitHub) getAnyLatestRelease(ctx context.Context) (*github.RepositoryRelease, error) {
+	release, _, err := g.client.Repositories.GetLatestRelease(ctx, g.owner, g.repo)
+	if err != nil {
+		// If the error is that no latest release was found, it could be that there are only pre-releases
+		if ghErrResp, ok := err.(*github.ErrorResponse); ok && ghErrResp.Response.StatusCode == http.StatusNotFound {
+			// Get the first release returned by ListReleases
+			releases, _, listErr := g.client.Repositories.ListReleases(ctx, g.owner, g.repo, &github.ListOptions{PerPage: 1})
+			if listErr != nil {
+				return nil, listErr
+			}
+			if len(releases) > 0 {
```

Comment:
Again, is it possible to not get an error and an empty list of releases? I would expect the API to return an 404 here as well, if there are no releases at all.